### PR TITLE
Right way to use system(3) function

### PR DESCRIPTION
**Pull request by leleliu008**

### CODE REVIEW

File: src/optimizer.cc [hunk 1]
```diff
@@ -523,12 +523,25 @@ Grammar Optimizer::optimize() {
...
-            int exit_code;
-            exit_code = WEXITSTATUS(system((debug_script + " " + filename).c_str()));
-            if (exit_code != 0) {
-                error("Debug script failed in pass %d! (exit_code=%d)", pass, exit_code);
+
+            int ret = system((debug_script + " " + filename).c_str());
+
+            if (ret == -1) {
+                perror(NULL);
+                error("Debug script failed in pass %d", pass);
```
Comment (dolik-rce) [inline src/optimizer.cc:531]: I think we could log `errno` here, to indicate why did the script failed. Something like 
```suggestion
                error("Debug script failed in pass %d: %s", pass, strerror(errno));
```

**Review by dolik-rce — CHANGES_REQUESTED** [review]
I assume that it is intended to fix the problem with compilation on macOS, right?

It looks good. There's just one little suggestion from me.

File: src/optimizer.cc (hunk 1 above)
Comment (leleliu008) [reply src/optimizer.cc:531]: I have invoked `perror(NULL);` before this line. Do you mean remove `perror(NULL);` and use `strerror(errno)`?

Comment (dolik-rce) [reply src/optimizer.cc:531]: Sorry, I've overlooked that. But yes, I think that printing both in one line is slightly more readable:

```
No such file or directory
ERROR: Debug script failed in pass 1
```
vs.
```
ERROR: Debug script failed in pass 1: No such file or directory
```

**Pull request merged**